msteams-platform/copilot/how-to-extend-copilot.md

```diff
@@ -0,0 +1,112 @@
+2. Provide a link to either an OpenAPI specification or an OpenAI plugin.
+3. Update your manifest with plugin metadata such as name, publisher, APIs used.
+4. Optional: If you want your plugin to return an Adaptive Card, provide an Adaptive Card template and reference the template in the manifest.
+5. Test your plugin.
+6. Preview and publish your plugin.
+
+**Scenario 2: I want to build a plugin from an existing Microsoft 365 message extension**
+
+Your Message Extension will work with Copilot (after quality testing).
+
+## What makes a good plugin for Microsoft 365 Copilot?
+
+To ensure your plugin works as intended, it is important to include good descriptions in your manifest. This allows the underlying LLM to take full advantage of your plugin’s capabilities. It’s recommended to test different descriptions and instructions to find the most effective way for the LLM to utilize your plugin. Also, it’s important that the manifest file is up-to-date with any changes or enhancements you make to your plugin.
+
+1. App description: Include detailed descriptions of the app, supported scenarios, feature capabilities, and related keywords using the `shortDescription` and `longDescription` fields.
+
+   ```json
+   ... 
+   "name": { 
+   "shortDescription": "Work-item tracking and productivity app", 
+   "longDescription": "Contoso app is a work-item tracking and project management app that allows teams to create, manage, and track work items. This app helps teams manage projects more efficiently. " 
+   ... 
+   } 
+   ```
+
+2. Skill parameter descriptions: Include detailed descriptions of your search and action skills along with associated parameters. These descriptions are used by the LLM to trigger the right skill and input the correct parameters for a given user request.
+
+   ```json
+    ... 
+   "commands": [{ 
+   "id": "Work item search query", 
+   "type": "query", 
+   "context": "command box", 
+   "parameters": [ 
+   { 
+   "name": "Work item search string", 
+   "title": "Work item search keyword", 
+   "description": "Search for work items by keyword(s)", 
+   }] 
+   }, 
+   ...
+   ```
+
+3. Adaptive Card responses: By incorporating Adaptive Cards into your plugin’s response, you can enhance the user experience by providing more dynamic, actionable content within the responses and references, allowing users to interact more effectively with the information provided.
+
+4. Including links in plugin response: Including links within your skill’s response serves multiple purposes:
+    * Assists the LLM in posting the entity link for users, streamlining the reference process.
+    * Enables users to easily access your entities in future Copilot requests.
+    * Embeds links in Adaptive Cards that allows your content to be displayed as a live, actionable loop component throughout the M365 ecosystem, enhancing user engagement and interaction.
```

Comment:
Should this say "embedded links" or is "embeds links" a technical term?

Reply:
Updatetd.